### ISSUE TITLE
Loop body counter never incremented for single-statement (no-block) loop bodies (`while`/`for`/`for-of`/`for-in`/`do-while`)

### ISSUE THREAD

**Issue by simleb** [fallow-rs/oxc-coverage-instrument#26]
When the body of a loop statement (`while`, `for`, `for-of`, `for-in`, `do-while`) is a single statement without a `{ … }` block, the body's `++cov.s[N]` counter is never emitted. The body's entry exists in `statementMap` but the statement is reported as 0-hit even when the loop runs.

Reproduces on `oxc-coverage-instrument@0.3.12`. Same family as the (closed) #19 if-without-else.

## Repro — `while` no-block

```js
import { instrument } from 'oxc-coverage-instrument'

const code = `function f(template) {
  const regex = /a/g
  const matches = []
  let match
  while ((match = regex.exec(template))) matches.push(match[0])
  return matches
}`

const r = instrument(code, 'x.js', { sourceMap: false, coverageVariable: '__cov__' })
new Function('exports', r.code.toString() + '; exports.f = f;')(globalThis.x = {})
console.log(globalThis.x.f('aaa'))   // ['a','a','a']  → body ran 3×

const cov = Object.values(globalThis.__cov__)[0]
console.log(cov.s)
// → { '0':1, '1':1, '2':1, '3':0, '4':1, '5':1 }
//                          ^ body counter still 0
```

The instrumented output for the loop line:

```js
while (match = regex.exec(template)) matches.push(match[0]);
//                                   ^^^^^^^^^^^^^^^^^^^^^ no `++cov.s[3]` here
```

There IS an entry `statementMap['3']` for the body's location range, but no corresponding increment is emitted in the transformed code, so it's permanently 0.

## All five loop variants are affected

```js
const sources = [
  // while
  `function f() { let i = 0; while (i < 3) i++; return i }`,
  // for
  `function f() { let total = 0; for (let i = 0; i < 3; i++) total++; return total }`,
  // for-of
  `function f(items) { let total = 0; for (const x of items) total++; return total }`,
  // for-in
  `function f(obj) { let total = 0; for (const k in obj) total++; return total }`,
  // do-while
  `function f() { let i = 0; do i++; while (i < 3); return i }`,
]
```

For each: `statementMap.size` is one greater than the number of `++cov.s[…]` increments the instrumenter emits — the body counter is the missing one.

## Expected

The single-statement body of a loop should get a `++cov.s[N]` injected at its start, just like a block-form body does:

```js
while (match = regex.exec(template)) {
  ++cov.s[N];        // already emitted for block form
  matches.push(match[0]);
}
```

Block-form bodies already increment correctly; only the no-block single-statement form is missing the injection.
